Add paragraph with description in default template

renatho opened the pull request:
Addresses https://github.com/Automattic/sensei/pull/5179#issuecomment-1141439880

### Changes proposed in this Pull Request

* It adds a paragraph to the default course template and fills it when skipping the editor wizard. [More context here](https://github.com/Automattic/sensei/pull/5179#issuecomment-1141439880)
* It renames all `"sensei-pattern-{X}"` to `"sensei-content-{X}"`, in order to unify the naming (now we have a placeholder in the CPT template besides the patterns).

### Testing instructions

<!--
If the functionality of this PR is behind a feature flag, please include the
following testing step, using the correct name for the feature flag. (If you
aren't sure, just ignore this step)

* Enable the feature flag: `add_filter( 'sensei_feature_flag_{THE_FLAG_NAME}', '__return_true' );`
-->
* Create a new course.
* Fill a description in the editor wizard.
* Close the editor wizard or click on "Start with default layout" in the patterns step.
* Make sure the description was added to the content.

<!--
Helpful tips for screenshots:
https://en.support.wordpress.com/make-a-screenshot/
-->
### Screenshot / Video

https://user-images.githubusercontent.com/876340/171956745-1c27cb06-15a5-408a-abf9-3872ba637197.mov

renatho committed:
Add paragraph description to default course template

renatho commented:
It depends on https://github.com/Automattic/sensei/issues/5213, so this is blocked for now since we'll need to refactor part of that code in order to also replace the content when skipping the wizard.

renatho committed:
Remove unused prop

renatho committed:
Lift state up

renatho committed:
Update current template with description when skipping wizard

renatho commented on assets/admin/editor-wizard/editor-wizard-modal.js:
```diff
@@ -3,83 +3,29 @@
 const EditorWizardModal = () => {
-	const [ open, setDone ] = useWizardOpenState();
-	const { synchronizeTemplate } = useDispatch( blockEditorStore );
+	const dataState = useState( {} );
```
It lifts up the data state from the wizard in order to have it available when skipping the wizard. We need the description to add to the post.

renatho commented on assets/admin/editor-wizard/editor-wizard-modal.js:
```diff
@@ -3,83 +3,29 @@
- *
- * @return {boolean|undefined} Whether a modal is open, or `undefined` if it's not initialized yet.
- */
-const useObserveOpenModal = ( shouldObserve ) => {
```
To simplify a little this component file, I moved these hooks to a `helpers.js` file.

fjorgemota replied:
Nice cleanup!

renatho committed:
Update hook name

renatho committed:
Add some tests to the open state wizard

renatho commented:
> Rename all "sensei-pattern-{X}" to "sensei-content-{X}", in order to unify the naming (now we have a placeholder in the CPT template besides the patterns).

It's still pending (a simple change) waiting for merges. Depending on the order of the merges, we can do it in another PR.

fjorgemota commented:
@renatho Considering that the base branch was merged, could you rebase the branch and rename the sensei-pattern-{X} to sensei-content-{X}?

I'll review this PR in the meantime =)

renatho committed:
Merge branch 'feature/course-lesson-wizard' of github.com:Automattic/sensei into add/paragraph-with-description-in-default-template

renatho committed:
Replace placeholder classes

We changed the name since it can also be a placeholder in the lesson CPT template

renatho commented:
> @renatho Considering that the base branch was merged, could you rebase the branch and rename the sensei-pattern-{X} to sensei-content-{X}?

✅

fjorgemota reviewed (approved):
Looks good and works well. :rocket:
